src/main/resources/assets/computercraft/lua/rom/apis/term.lua

```diff
@@ -12,17 +12,15 @@ local term = {}
 
 term.redirect = function( target )
 	if target == nil or type( target ) ~= "table" then
-		error( "Invalid redirect target", 2 )
+		error( "bad argument #1 (expected table, got " .. type( target ) .. ")", 2 )
 	end
     if target == term then
         error( "term is not a recommended redirect target, try term.current() instead", 2 )
     end
 	for k,v in pairs( native ) do
 		if type( k ) == "string" and type( v ) == "function" then
 			if type( target[k] ) ~= "function" then
-				target[k] = function()
-					error( "Redirect object is missing method "..k..".", 2 )
-				end
+				error( "Redirect object is missing method "..k..".", 2 )
```

Comment:
I don't think it is.

Current behaviour is backwards compatible with older buffers, which may not offer functions such as the relatively recent [term.blit()](http://www.computercraft.info/wiki/Term.blit). Since these older buffers tend to be used alongside older scripts which won't call that function anyway, they don't run into any problems.

This change would forcibly break them, and I don't see any gain.